### ISSUE TITLE
Switch to valid URL syntax

### ISSUE THREAD

**Issue by ara4n** [matrix-org/matrix.to#17]
Technically you're meant to escape # and @ symbols within URL paths/fragments. The fact we don't causes woes like those described on https://meta.discourse.org/t/broken-links-blank-page-with-appearing-in-url/52640/10

**Comment by ara4n** [comment]
The solution to this is probably to switch over to compact URLs everywhere as per issue #10.

**Comment by uhoreg** [comment]
The problem with the compact URLs is that a URL like matrix.to/@matthew:matrix.org would 1) send the entire URL to the server, which may be a privacy concern, and 2) require some server-side help, which would make it harder for people to host it themselves (e.g. matrix.to currently 404s on that link instead of loading the main matrix.to page)

**Comment by StefanBrand** [comment]
Usually one would handle it like this, I guess: 

`https://matrix.to?r=matrix:matrix.org`

`https://matrix.to?u=Matthew:matrix.org`

**Comment by ara4n** [comment]
yeah, but i'd prefer it to look more cosmetically like a mx://server/resource URL rather than a slightly clunky querystring, and to preserve the correct sigil semantics of matrix user ids, room aliases etc (even though they don't fit very nicely into http urls sadly)

**Comment by phaux** [comment]
How about `/u/user/example.com` for users and `/r/room/example.com` for rooms?

**Comment by t3chguy** [comment]
@phaux what about roomId `!` vs roomAlias `#` and also event permalinks `!`/`$`

**Comment by eMPee584** [comment]
🥁

**Comment by skepticalwaves** [comment]
Another big problem with the "#" syntax is that this makes the matrix.to implementation impossible to proxy with apache/nginx as the full URL doesn't get sent as part of the "GET".

https://stackoverflow.com/questions/9484852/hash-character-in-urls-accessing-and-redirecting-in-apache

**Comment by t3chguy** [comment]
> as the full URL doesn't get sent as part of the "GET".

This is by design. To prevent the host being able to sniff on usage.
